developing/docker/README.md

```diff
@@ -94,6 +98,162 @@ To get a `top` like report of what your containers are doing
 
     docker stats
 
+## Image IDs, Tags, and Digests
+
+Docker image identifiers and SHA256 digests (aka hashes or checksums) are a
+complex topic. In the context of our work at Truss, the topic is especially
+relevant when figuring out which value to use when referencing an image in an
+Amazon Elastic Container Service (ECS) task definition.
+
+### Problem Statement and Summary
+
+An [ECS task
+definition](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_image)
+can pull an image by tag or digest, so which should you use? Image tags are
+potentially mutable and are not tied to the image contents. For instance, a new
+image could be created with different contents, but could reuse the same tag as
+an earlier, different image. If your ECS task definition pulls the image based
+on tag, you could be pulling in a different image than expected. An image's
+digest, on the other hand, is a hash based on the image contents. Pulling the
+image based on digest provides greater security and protection against
+unintended changes.
+
+When building a Docker image locally, the SHA256 digest to use in the ECS task
+definition won't be known until the image has been pushed to the Docker
+registry. The reason for this is explained in more detail in the sections
+below.
+
+However, once the image has been pushed to the registry, the SHA256 digest to
+use in the ECS task definition can be found with the following command:
+
+    docker image inspect some_image_name -f '{{.RepoDigests}}'
```

Comment:
Note that `RepoDigests` is an array, so if you have perchance pushed to more than one repository, you will need to make sure you get the digest for the right repository.